boltz/db.go

```diff
@@ -198,38 +222,63 @@ func (self *DbImpl) RootBucket(tx *bbolt.Tx) (*bbolt.Bucket, error) {
 	return rootBucket, nil
 }
 
-func (self *DbImpl) Snapshot(tx *bbolt.Tx) error {
-	self.reloadLock.RLock()
-	defer self.reloadLock.RUnlock()
-
+func (self *DbImpl) GetDefaultSnapshotPath() string {
 	path := self.db.Path()
 	path += "-" + time.Now().Format("20060102-150405")
+	return path
+}
 
-	_, err := os.Stat(path)
+func (self *DbImpl) Snapshot(path string) (string, string, error) {
+	var actualPath string
+	var snapshotId string
+	err := self.View(func(tx *bbolt.Tx) error {
+		var err error
+		actualPath, snapshotId, err = self.SnapshotInTx(tx, path)
+		return err
+	})
 	if err != nil {
-		if !os.IsNotExist(err) {
-			return err
-		}
-	} else {
-		pfxlog.Logger().Infof("bolt db backup already made: %v", path)
-		return nil
+		return "", "", err
 	}
+	return actualPath, snapshotId, nil
+}
 
-	file, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE, 0600)
-	if err != nil {
-		return err
+func (self *DbImpl) SnapshotInTx(tx *bbolt.Tx, path string) (string, string, error) {
+	self.reloadLock.RLock()
+	defer self.reloadLock.RUnlock()
+
+	now := time.Now()
+	dateStr := now.Format("20060102")
+	timeStr := now.Format("150405")
+	path = strings.ReplaceAll(path, "DATE", dateStr)
+	path = strings.ReplaceAll(path, "TIME", timeStr)
+	path = strings.ReplaceAll(path, "DB_DIR", filepath.Dir(self.db.Path()))
+	path = strings.ReplaceAll(path, "DB_FILE", filepath.Base(self.db.Path()))
```

Comment:
People could have folders called DATE or TIME or even partials.

Should this be in curly braces or some other sentinel token?

Reply:
It's the established convention from the ziti snapshot functionality, I just moved it down a layer. So if I change it, then it's not backward compatible. 
I could add additional variables and mark these deprecated. Since they're provided on the command line, we'd want something that wont' get interpreted by a shell. Maybe something like __DATE__? 
Thougts?

Reply:
Outside of PR conversation, landed on using double underscore pre and postfix.